app/controllers/host/vms_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Host::VmsController < ApplicationController
+  before_action :find_compute_resource, :only => :new
+  before_action :find_vm, :only => :new
+
+  helper :hosts
+
+  # Import a host from a compute resource
+  def new
```

Comment:
I think, GET is correct here as it - as you already said - it just renders a host edit form with predefined values. The actual creation is happening like for every other host. In discovery, we do something similar.

The API call is just a simple create. The only difference is, that when you pass `uuid` and `compute_resource_id` Foreman doesn't create a new VM but links the existing one.